### PR TITLE
Add pytest to the CI pipeline

### PR DESCRIPTION
**Pull request by pozsa**
Trigger the workflow only if there are changes to backend, but ignore
changes to the README inside.

See: #34

### CODE REVIEW

File: .github/workflows/test-backend.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install poetry
+          poetry config virtualenvs.create false
+          poetry install --no-interaction --no-ansi
+
+      - name: Run tests
+        run: |
+          pytest
```
Comment (minottic) [inline .github/workflows/test-backend.yaml:38]: I think this will need to change, based on yesterday's discussion (or we should commit the __init__)

**Review by minottic — COMMENTED** [review]
I thought the tests workflow would run in this PR as well, do you know why it didn't?

File: .github/workflows/test-backend.yaml (hunk 1 above)
Comment (pozsa) [reply .github/workflows/test-backend.yaml:38]: Because of the workflow trigger definition. (the `on.paths` section)

Comment (pozsa) [reply .github/workflows/test-backend.yaml:38]: We said we will just mark the `tests` dir as a regular pkg with an `__init__.py`. That way this line doesn't have to change.

Comment (minottic) [reply .github/workflows/test-backend.yaml]: understood, thanks. Do we want changes in the `workflow/test-backend.yaml` file to trigger a CI?

Comment (pozsa) [reply .github/workflows/test-backend.yaml]: To make sure we are not breaking the testing completely. Yeah, that makes sense.

Comment (pozsa) [reply .github/workflows/test-backend.yaml]: @minottic it fails atm, because there are no tests to run. (exit code 5)

Comment (minottic) [reply .github/workflows/test-backend.yaml]: I was about to write the same comment :). I think we can merge it as it is

**Review by minottic — APPROVED** [review]

**Pull request merged**